src/server/index.ts

```diff
@@ -9,28 +9,56 @@ import bodyParser from 'body-parser';
 import { configureStore } from '../shared/store';
 import serverRender from './render';
 import paths from '../../config/paths';
+import { ErrorRequestHandler, Handler, RequestHandler, Request } from 'express-serve-static-core';
+import { Store } from 'redux';
 
 require('dotenv').config();
 
 const app = express();
 
+interface RequestWithStore extends Request {
+    store?: Store;
+}
+
+const errorHandler: ErrorRequestHandler = (err, _, res) =>
+    res.status(404).json({
+        status: 'error',
+        message: err.message,
+        stack:
+            // print a nicer stack trace by splitting line breaks and making them array items
+            process.env.NODE_ENV === 'development' &&
+            (err.stack || '')
+                .split('\n')
+                .map((line: string) => line.trim())
+                .map((line: string) => line.split(path.sep).join('/'))
+                .map((line: string) =>
+                    line.replace(
+                        process
+                            .cwd()
+                            .split(path.sep)
+                            .join('/'),
+                        '.'
+                    )
+                ),
```

Comment:
Lines 28 to 42 make my eyes burn. At least they should go in a separate function.